utils/s3/client.cc

```diff
@@ -112,7 +128,8 @@ shared_ptr<client> client::make(std::string endpoint, endpoint_config_ptr cfg, s
 }
 
 future<> client::authorize(http::request& req) {
-    if (!_credentials || std::chrono::system_clock::now() >= _credentials.expires_at) {
+    if (!_credentials) {
+        s3l.info("Update creds synchronously");
         _credentials = co_await _creds_provider_chain.get_aws_credentials();
```

Comment:
If multiple requests come at this place we end up having a bunch of get_aws_credentials() running in parallel